samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs

```diff
@@ -10,70 +13,38 @@
 using Avro.File;
 using Avro.Generic;
 using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.EventGrid;
 using Microsoft.Azure.WebJobs.Extensions.Http;
 using Microsoft.Azure.WebJobs.Host;
 using Microsoft.WindowsAzure.Storage;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace FunctionEGDWDumper
+{
 
-namespace FunctionDWDumper
-{    
-    public static class DWDumperFunction1
+    public static class Function1
     {
         private static readonly string StorageConnectionString = Environment.GetEnvironmentVariable("StorageConnectionString");
         private static readonly string SqlDwConnection = Environment.GetEnvironmentVariable("SqlDwConnection");
 
         /// <summary>
         /// Use the accompanying .sql script to create this table in the data warehouse
-        /// </summary>
+        /// </summary>  
         private const string TableName = "dbo.Fact_WindTurbineMetrics";
 
-        /// <summary>
-        /// Before wiring this up with EventGrid, you can test this function by
-        /// a. Create a new EventGrid subscription name to https://requestb.in
-        ///    This keeps sending EventGrid json data to https://requestb.in
-        /// b. From there, copy the json request body
-        /// c. Execute the function in Azure portal as a one-off by selecting POST and providing the json body you copied
-        /// d. This should create some rows in the data warehouse.
-        ///    This ensures that the Azure Function is dumping data correctly to the data warehouse
-        /// e. Next, create a new EventGrid subscription name to Azure Functions
-        ///    This should automatically populate data to the data warehouse every time the Avro blob is created by Event Hubs Capture.
-        /// </summary>
-        /// <returns></returns>
-        [FunctionName("HttpTriggerCSharp")]
-        public static async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequestMessage req, TraceWriter log)
+        [FunctionName("Function1")]
+        public static void Run([EventGridTrigger]JObject eventGridEvent, TraceWriter log)
```

Comment:
Also, can we add comments describing the steps? please who download will get better perspective (refer to the previous code)

Reply:
As a best practice, lets add Async methods and exception handling as well